velox/expression/CastExpr.cpp

```diff
@@ -246,7 +246,12 @@ VectorPtr applyDecimalToIntegralCast(
   const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
   const auto scaleFactor = DecimalUtil::kPowersOfTen[precisionScale.second];
   context.applyToSelectedNoThrow(rows, [&](int row) {
-    int128_t integralPart = simpleInput->valueAt(row) / scaleFactor;
+    int128_t integralPart = simpleInput->valueAt(row) / (scaleFactor >> 1);
```

Comment:
This will result in a divide by 0 error if the scaleFactor is 1 or 0.
You use the remainder to roundUp.

Reply:
May be try to use this function `DecimalUtill::divideWithRoundUp`. https://github.com/facebookincubator/velox/blob/main/velox/type/DecimalUtil.h#L151